### PR TITLE
Improve presto-druid escaping special chars in column and table name 

### PR DESCRIPTION
**Pull request by beinan**
1 escaping special characters in the column name
2 escaping special characters in the table name
3 escaping special characters for the json request to druid server
4 fix bugs for extra escaping when scan hdfs file. 
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by zhenxiao** [comment]
[ERROR] testFilterExpressionConverter(com.facebook.presto.druid.TestDruidExpressionConverters)  Time elapsed: 0.079 s  <<< FAILURE!
java.lang.IllegalArgumentException: no type found found for symbol 'regionid'
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:216)
	at com.facebook.presto.sql.planner.TypeProvider.get(TypeProvider.java:64)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer$Visitor.visitSymbolReference(ExpressionAnalyzer.java:390)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer$Visitor.visitSymbolReference(ExpressionAnalyzer.java:313)
	at com.facebook.presto.sql.tree.SymbolReference.accept(SymbolReference.java:41)
	at com.facebook.presto.sql.tree.StackableAstVisitor.process(StackableAstVisitor.java:26)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer$Visitor.process(ExpressionAnalyzer.java:336)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer$Visitor.getOperator(ExpressionAnalyzer.java:1274)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer$Visitor.visitComparisonExpression(ExpressionAnalyzer.java:494)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer$Visitor.visitComparisonExpression(ExpressionAnalyzer.java:313)
	at com.facebook.presto.sql.tree.ComparisonExpression.accept(ComparisonExpression.java:71)
	at com.facebook.presto.sql.tree.StackableAstVisitor.process(StackableAstVisitor.java:26)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer$Visitor.process(ExpressionAnalyzer.java:336)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer.analyze(ExpressionAnalyzer.java:279)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer.analyzeExpressions(ExpressionAnalyzer.java:1525)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer.getExpressionTypes(ExpressionAnalyzer.java:1507)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer.getExpressionTypes(ExpressionAnalyzer.java:1494)
	at com.facebook.presto.sql.analyzer.ExpressionAnalyzer.getExpressionTypes(ExpressionAnalyzer.java:1481)
	at com.facebook.presto.druid.TestDruidQueryBase.toRowExpression(TestDruidQueryBase.java:188)
	at com.facebook.presto.druid.TestDruidQueryBase.getRowExpression(TestDruidQueryBase.java:212)
	at com.facebook.presto.druid.TestDruidExpressionConverters.testFilter(TestDruidExpressionConverters.java:79)
	at com.facebook.presto.druid.TestDruidExpressionConverters.testFilterExpressionConverter(TestDruidExpressionConverters.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestDruidExpressionConverters.testFilterExpressionConverter:55->testFilter:79->TestDruidQueryBase.getRowExpression:212->TestDruidQueryBase.toRowExpression:188 Â» IllegalArgument
[INFO] 
[ERROR] Tests run: 14, Failures: 1, Errors: 0, Skipped: 0

**Comment by beinan** [comment]
unit test passed @zhenxiao

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java [hunk 1]
```diff
@@ -183,4 +185,44 @@ public InputStream handle(Request request, com.facebook.airlift.http.client.Resp
             }
         }
     }
+    public static class DruidRequestBody
```
Comment (zhenxiao) [inline presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:188]: private?

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java [hunk 2]
```diff
@@ -183,4 +185,44 @@ public InputStream handle(Request request, com.facebook.airlift.http.client.Resp
...
+    {
+        private String query;
+        private String resultFormat;
+        private boolean queryHeader;
```
Comment (zhenxiao) [inline presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:192]: add blank line below

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java [hunk 3]
```diff
@@ -183,4 +185,44 @@ public InputStream handle(Request request, com.facebook.airlift.http.client.Resp
...
+        @JsonCreator
+        public DruidRequestBody(
+                @JsonProperty("query") String query,
+                @JsonProperty("resultFormat") String resultFormat,
+                @JsonProperty("queryHeader") boolean queryHeader)
+        {
+            requireNonNull(query);
```
Comment (zhenxiao) [inline presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:199]: this.query = requireNonNull(query);

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java [hunk 4]
```diff
@@ -183,4 +185,44 @@ public InputStream handle(Request request, com.facebook.airlift.http.client.Resp
...
+            this.query = query;
+            this.resultFormat = resultFormat;
```
Comment (zhenxiao) [inline presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:201]: this. resultFormat = requireNonNull(resultFormat);

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java [hunk 5]
```diff
@@ -168,7 +168,7 @@ public DruidQueryGeneratorContext withAggregation(
                     String definition = entry.getValue().getDefinition();
                     int start = definition.indexOf("(");
                     int end = definition.indexOf(")");
-                    String countDistinctClause = "count ( distinct " + definition.substring(start + 1, end) + ")";
+                    String countDistinctClause = "count ( distinct \"" + definition.substring(start + 1, end) + "\")";
```
Comment (zhenxiao) [inline presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java:171]: create a private function to add quotes to column names. could be used here and below

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java [hunk 6]
```diff
@@ -244,14 +244,14 @@ private boolean hasAggregation()
...
-                .map(s -> s.getValue().getDefinition())
+                .map(s -> s.getValue().getEscapedDefinition())
                 .collect(Collectors.joining(", "));
         if (expressions.isEmpty()) {
             throw new PrestoException(DRUID_QUERY_GENERATOR_FAILURE, "Empty Druid query");
         }
 
         String tableName = from.orElseThrow(() -> new PrestoException(DRUID_QUERY_GENERATOR_FAILURE, "Table name missing in Druid query"));
-        String query = "SELECT " + expressions + " FROM " + tableName;
+        String query = "SELECT " + expressions + " FROM \"" + tableName + "\"";
```
Comment (zhenxiao) [inline presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java:254]: private function to add quotes to table or column names. could be used here

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java (hunk 1 above)
Comment (beinan) [reply presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:188]: > private?

Yep

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java (hunk 2 above)
Comment (beinan) [reply presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:192]: added

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java (hunk 3 above)
Comment (beinan) [reply presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:199]: done

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java (hunk 4 above)
Comment (beinan) [reply presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:201]: done

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java (hunk 6 above)
Comment (beinan) [reply presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java:254]: done

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java (hunk 5 above)
Comment (beinan) [reply presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java:171]: done

File: presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java (hunk 3 above)
Comment (zhenxiao) [reply presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java:200]: remove line 200?

**Pull request merged**